src/core/load-fragment.c

```diff
@@ -362,6 +363,30 @@ int config_parse_unit_string_printf(
         return config_parse_string(unit, filename, line, section, section_line, lvalue, ltype, k, data, userdata);
 }
 
+int config_parse_reboot_parameter(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        assert(filename);
+        assert(line);
+        assert(rvalue);
+
+        if (!reboot_parameter_is_valid(rvalue)) {
+                log_syntax(unit, LOG_WARNING, filename, line, SYNTHETIC_ERRNO(EINVAL), "Invalid reboot parameter '%s', ignoring.", rvalue);
+                return 0;
+        }
+
+        return config_parse_unit_string_printf(unit, filename, line, section, section_line, lvalue, ltype, rvalue, data, userdata);
```

Comment:
please use unit_full_printf() on the string, before you test its validity, not after.

otherwise you'll say a string ok which contains all kind of weird stuff after specifiers get expandend.